### PR TITLE
Implemented #197

### PR DESCRIPTION
**Pull request by felixschorer**
Ok, this looks like a huge change, but it isn't. Calm down 😎 

- Did some renaming and moving of files. `master-process.ts` shows as a new file despite only being renamed from `processing-manager.ts` with some added changes.

- Redefined parameter logging, you were right after all @nyxathid 

- Made queue ops scripts part of our application

- Updated README. I **highly** suggest you read it, seriously! The usage has somewhat changed

- Also finally removed `TestRuns` as it was only testing legacy code anyways.

### CODE REVIEW

**Comment by felixschorer** [comment]
Wow, why the f*ck has yarn install to fail during the CI test...

**Comment by felixschorer** [comment]
Forgot to implement a CLI parameter...

File: src/utils/cli.ts [hunk 1]
```diff
@@ -60,76 +50,101 @@ export class CLI {
...
         if (this.commander.blobLocation) {
-            if (!this.commander.blobLocation.split) {
+            let split = this.commander.blobLocation.split(":", 2);
+            if (split.length < 2) {
                 console.warn("invalid --blob-location [account]:[container]");
             } else {
-                let splitted = this.commander.blobLocation.split(":", 2);
-                this.parameters.blobAccount = splitted[0];
-                this.parameters.blobContainer = splitted[1];
+                this.parameters["blobAccount"] = split[0];
```
Comment (nyxathid) [inline src/utils/cli.ts]: Just curious, what is the difference between `["blobAccount"]` and `.blobAccount`?

File: src/utils/param-loader.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        crawlVersion: "CC-MAIN-2017-13",
+        heuristicThreshold: 3,
+        languageCodes: ["en"]
+    };
+
+    private constructor() {
+        this.CLI = CLI.getInstance().parameters;
+        this.config = ParamLoader.loadConfig();
+        this.env = ParamLoader.parseEnvVars();
+        this.all = Object.assign({}, this.DEFAULT, this.env, this.config, this.CLI);
```
Comment (nyxathid) [inline src/utils/param-loader.ts]: CLI overrides config overrides env overrides dafault. Do I get it right?

File: README.md [hunk 3]
```diff
@@ -1,32 +1,75 @@
...
-$ node out/app.js -h
-
-> Usage: app [options]
+$ Usage: app [options]
 
-  Options:
+> Options:
 
     -h, --help                                 output usage information
+    -P, --Process                              process queue items
```
Comment (nyxathid) [inline README.md]: Are `-A`, `-M` and `-P` the "big" modes? Can yo run them simultaneously?

File: src/utils/cli.ts (hunk 1 above)
Comment (felixschorer) [inline src/utils/cli.ts]: `.blobAccount` requires you to define the object structure in TS, `["blobAccount"]` does not. 
Until now we had this big object with all its properties set to undefined. I had to change this as `ParamLoader` will fail to merge the parameters when they are defined, but set to undefined.

File: src/utils/param-loader.ts (hunk 2 above)
Comment (felixschorer) [inline src/utils/param-loader.ts]: Exactly! The first parameter of `Object.assign` will be mutated. That's why there is an empty object.

File: README.md (hunk 3 above)
Comment (felixschorer) [inline README.md]: Yes, indeed. You have to specify at least one, but you can also s all 3.

**Comment by felixschorer** [comment]
All of these changes have been extensively tested while doing tests for #196

File: README.md (hunk 3 above)
Comment (felixschorer) [inline README.md]: Using both `-A` and `-P` would replicate the functionality of the version where we didn't have the Azure queue.

File: src/utils/param-loader.ts (hunk 2 above)
Comment (nyxathid) [inline src/utils/param-loader.ts]: cool stuff!

**Pull request merged**